packages/flutter/lib/src/material/carousel_theme.dart

```diff
@@ -0,0 +1,185 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+import 'carousel.dart';
+import 'color_scheme.dart';
+import 'material.dart';
+import 'theme.dart';
+
+/// Defines default property values for descendant [CarouselView] widgets.
+///
+/// Descendant widgets obtain the current [CarouselViewThemeData] object using
+/// `CarouselTheme.of(context)`. Instances of [CarouselViewThemeData] can be
+/// customized with [CarouselViewThemeData.copyWith].
+///
+/// Typically a [CarouselViewThemeData] is specified as part of the overall [Theme]
+/// with [ThemeData.carouselViewTheme].
+///
+/// All [CarouselViewThemeData] properties are `null` by default. When null, the [CarouselView]
+/// will provide its own defaults.
+///
+/// See also:
+///
+///  * [CarouselViewTheme], an [InheritedWidget] that propagates the theme to its descendants.
+///  * [ThemeData], which describes the overall theme information for the application.
+@immutable
+class CarouselViewThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.carouselViewTheme].
+  const CarouselViewThemeData({
+    this.elevation,
+    this.backgroundColor,
+    this.overlayColor,
+    this.shape,
+    this.padding,
+  });
+
+  /// The padding between the carousel's edge and its children.
```

Comment:
This doc looks a bit misleading. I think padding is the amount of space to surround each carousel item with. We might also want to mention (like other component theme doc), "Overrides the default value for [CarouselView.padding]". Here and the properties below.